source/util/Retry.cpp

```diff
@@ -10,38 +10,58 @@ using namespace Aws::Iot::DeviceClient::Util;
 
 const char *Retry::TAG = "Retry.cpp";
 
-bool Retry::exponentialBackoff(function<bool()> retryableFunction, ExponentialRetryConfig config)
+bool Retry::exponentialBackoff(
+    function<bool()> retryableFunction,
+    function<void()> onComplete,
+    ExponentialRetryConfig config)
 {
+    bool needToStop = false;
+    if (config.needStopFlag != nullptr)
+    {
+        needToStop = config.needStopFlag->load();
+    }
+
+    if (needToStop)
+    {
+        LOG_DEBUG(TAG, "Stop flag was set prior to executing retryable function, will not attempt retryable execution");
+        return false;
```

Comment:
Should onComplete be called here?

Reply:
I think that's a good idea! I will add it